### PR TITLE
Silence various warnings on GCC 4.9.3

### PR DESCRIPTION
**Pull request by antoinealb**
When building latest Aseba master on ARM Cortex-M several warnings are produced. This PR fixes all of them and should not induce any behaviour changes.

### CODE REVIEW

File: transport/can/can-buffer.c [hunk 1]
```diff
@@ -23,12 +23,14 @@
 
 void AsebaSendBuffer(AsebaVMState *vm, const uint8* data, uint16 length)
 {
+	(void) vm;
```
Comment (stephanemagnenat) [inline transport/can/can-buffer.c]: I guess this aims at removing a "variable not used" warning. I was not aware about this usage before this pull request, do you have a link to a reference/rationale for it? I'm curious!

Comment (antoinealb) [inline transport/can/can-buffer.c]: You are correct: this prevents warnings about unused variables. I do not have a "serious" reference available but I have seen it used on several open source projects. See also this SO answer: http://stackoverflow.com/a/308286

**Comment by stephanemagnenat** [comment]
I have added an ASEBA_UNUSED macro in commit 4b4cbc522f8da83269ad20e606665a10ce5f05a3, could you please update the pull request to use it? It is more informative than using (void) directly.

**Comment by antoinealb** [comment]
Sure. I rebased my PR and added `ASEBA_UNUSED`.

File: transport/can/can-buffer.c [hunk 2]
```diff
@@ -20,15 +20,18 @@
 
 #include "../buffer/vm-buffer.h"
 #include "can-net.h"
+#include "common/consts.h"
```
Comment (stephanemagnenat) [inline transport/can/can-buffer.c]: This path should be relative, "../../common/consts.h"

Comment (antoinealb) [inline transport/can/can-buffer.c]: Ok, wasn't aware of that. Fixed now.

**Pull request merged**